labs/LanguageBasics/LanguageBasics.md

```diff
@@ -211,60 +625,137 @@ encrypt key plainText = cipherText
+Using exhaustive testing.
+Passed 16 tests.
+Q.E.D.
+```
+
+The `:check` command is smart enough to notice small enough domains
+and switch to exhaustion automagically:
+
+```shell
+labs::LanguageBasics::LanguageBasics> :check add8 === sub8
+Using exhaustive testing.
+Passed 16 tests.
+Q.E.D.
+```
+
+
+Judicious Type System Usage
+---------------------------
+
+### Don't let the type system do your work
+
+Cryptol's type system tries to infer the types of functions lacking a
+type signature. Sometimes it comes up with a more general type than
+you were imagining. This causes problems:
+
+* Perhaps you only want your function to be applicable on a smaller
+  set of types (usually minor, but occasionally major).
+* Error messages can become even more incomprehensible! (Major)
+
+### Do let the type system work for you
+
+Type signatures for functions are wonderful bits of documentation. It
+is much easier to see what's going on if you use type synonyms and
+signatures.
 
 * makes cleaner code
 * easier for other tools to consume/reason about
 
-## Supporting code
+### Provide additional types to aid in debugging
+
+Many of the errors in coding Cryptol will be instances of type
+mismatching. If you can't see your problem based on the error message,
+try adding more type annotations. This
+* makes the interpreter do less work trying alternative possibilites
+and, consequently, can make error messages more comprehensible
+* reduces the body of code to examine for bugs (a sort of binary bug
+search)
+* can get you to notice where you blew it
+
+
+Here Abide Monsters
+-------------------
+
+Following is some code that is needed to make the Cryptol within this
+document valid, but is not discussed for pedagogical reasons. You may
+read on, but you have been warned...
 
-Following is some code that supports this lab but is not discussed for pedagogical reasons. It does serve to illustrate the type signature and function definitions can be separated within a file. A practice that is **strongly** discouraged. 
+The following code does serve to illustrate the type signature and
+function definitions can be separated within a file. A practice that
+is _**strongly**_ discouraged.
 
 ```
-gcdCurried = gcd  
+gcdCurried = gcd
 gcdUncurried = uncurry gcdCurried
 ```
 
-These illustrate higher order functions. We define the _function_ `gcdCurried` in terms of the _function_ `gcd` without mentioning arguments. We use the built in `uncurry` higher-order function which takes a two argument curried function and returns an uncurried version (a one argument function operating on an ordered pair).
+These also illustrate higher order functions. We define the _function_
+`gcdCurried` in terms of the _function_ `gcd` without mentioning
+arguments. We use the built in `uncurry` higher-order function which
+takes a two argument curried function and returns an uncurried version
+(a one argument function operating on an ordered pair).
```

Comment:
Care to add a closing comment?  A denouement, if you will?  ;-)

Reply:
Added in postface.